tests/contrib/test_kmeans.py

```diff
@@ -0,0 +1,107 @@
+        assert out1.shape == (N,)
+        assert out2.shape == (num_clusters, D)
+
+    def test_exception(self, device, dtype):
+        # case: cluster_center = 0:
+        with pytest.raises(Exception) as errinfo:
+            kornia.contrib.KMeans(0, None, 10e-4, 10, 0)
+        assert "num_clusters can't be 0" in str(errinfo)
+
+        # case: cluster centers is not a 2D tensor
+        with pytest.raises(TypeError) as errinfo:
+            starting_centers = torch.rand((2, 3, 5), device=device, dtype=dtype)
+            kmeans = kornia.contrib.KMeans(None, starting_centers, 10e-4, 100, 0)
+        assert "shape must be [['C', 'D']]. Got torch.Size([2, 3, 5])" in str(errinfo)
+
+        # case: input data is not a 2D tensor
+        with pytest.raises(TypeError) as errinfo:
+            kmeans = kornia.contrib.KMeans(3, None, 10e-4, 100, 0)
+            kmeans.fit(torch.rand((1000, 5, 60), dtype=dtype, device=device))
+        assert "shape must be [['N', 'D']]. Got torch.Size([1000, 5, 60])" in str(errinfo)
+
+        # case: column dimensions of cluster centers and data to be predicted do not match
+        with pytest.raises(Exception) as errinfo:
+            kmeans = kornia.contrib.KMeans(3, None, 10e-4, 100, 0)
+            kmeans.fit(torch.rand((1000, 5), dtype=dtype))
+            kmeans.predict(torch.rand((10, 7), dtype=dtype))
+        assert "7 != 5" in str(errinfo)
+
+    def test_module(self, device, dtype):
+        # create example dataset
+        torch.manual_seed(2023)
+        x = 5 * torch.randn((500, 2), dtype=dtype, device=device) + torch.tensor((-13, 17), dtype=dtype, device=device)
+        x = torch.vstack(
+            [x, torch.randn((500, 2), dtype=dtype, device=device) + torch.tensor((15, -12), dtype=dtype, device=device)]
+        )
+        x = torch.vstack(
+            [
+                x,
+                13 * torch.randn((500, 2), dtype=dtype, device=device)
+                + torch.tensor((35, 15), dtype=dtype, device=device),
+            ]
+        )
+
+        kmeans = kornia.contrib.KMeans(3, None, 10e-4, 10000, 2023)
+        kmeans.fit(x)
+
+        centers = kmeans.get_cluster_centers()
+        prediciton = kmeans.predict(torch.tensor([[-14, 16], [45, 12]], dtype=dtype, device=device)).tolist()
+
+        expected_centers = torch.tensor([[-13, 17], [15, -12], [35, 15]], dtype=dtype, device=device)
+        expected_prediciton = [0, 2]
+
+        # sorting centers using dimension 0 as key so that they can be checked for equalness
+        order = torch.argsort(centers[:, 0]).tolist()
+        new_classes = {old_class: new_class for new_class, old_class in enumerate(order)}
+
+        ordered_centers = centers[order]
+        oredered_prediciton = [new_classes[predicted_class] for predicted_class in prediciton]
+
+        self.assert_close(ordered_centers, expected_centers, atol=2, rtol=0.1)
+        assert oredered_prediciton == expected_prediciton
+
+    def test_gradcheck(self, device):
```

Comment:
not sure we need this one here ? worth to add though `torchscript/dynamo` tests

Reply:
I'm not sure about what dynamo is. Where can I read about it? From other test cases it seems like we make an optimised version of the op and test the module again.

Reply:
Hi @avinashselvam, dynamo is the [`torch.compile` API](https://pytorch.org/tutorials/intermediate/torch_compile_tutorial.html)

Here is an example of how we are doing in our tests

https://github.com/kornia/kornia/blob/e40da282ba84b02ed8f3958dc220ce76d20572aa/tests/color/test_gray.py#L91